rfd/0204-aws-iam-roles-anywhere.md

```diff
@@ -0,0 +1,769 @@
+          - arn:aws:iam::123456789012:role/my-custom-role3
+```
+
+Teleport must check if the Profile Roles are compatible with the Integration's Trust Anchor.
+This is done using the `iam:GetRole` and checking its trust policy property:
+```json
+        "Condition": {
+          "ArnEquals": {
+            "aws:SourceArn": [
+              "<TrustAnchorARN>"
+            ]
+          }
+        }
+```
+
+### AWS Roles Anywhere Certificate Authority
+
+A new Certificate Authority will be created in Teleport to issue X.509 certificates, consumed by IAM Roles Anywhere.
+
+This new CA is backed by a ECDSA key, according to our current recommended crypto suites (see RFD0136).
+
+#### CA certificate
+When setting the trust anchor in AWS IAM Roles Anywhere, we'll add the new CA certificate.
+
+This certificate must comply with the following:
+- must be a X.509v3 certificate
+- Basic Constraints must include `CA: true`
+- the signing algorithm must include SHA256 or stronger
+- key usage must include Certificate Sign, CRL Sign and Digital Signature
+
+Including the CRL Sign is not required at this point, but this allows us to implement certificate revocation later on without the need to change the Trust Anchor.
+
+[See more](https://docs.aws.amazon.com/rolesanywhere/latest/userguide/trust-model.html#signature-verification)
+
+#### End entity certificates - X.509
+When issuing X.509 certificates for users to access AWS, they must comply with the following:
+- must be a X.509v3 certificate
+- Basic Constraints must not include `CA` or its value must be false
+- the signing algorithm must include SHA256 or stronger
+- key usage must include Digital Signature
+
+[See more](https://docs.aws.amazon.com/rolesanywhere/latest/userguide/trust-model.html#signature-verification)
+
+Sessions generated from IAM Roles Anywhere certificates have their [Source Identity](https://docs.aws.amazon.com/IAM/latest/UserGuide/id_credentials_temp_control-access_monitor.html) set to the common name (CN) of the certificate's Subject.
+Given its importance for auditing, this should contain the Teleport User name that requested the certificate.
+
+During IAM policy evaluation, the following certificate fields are exposed and can be used to allow or deny the request:
+- Subject fields as `PrincipalTag: aws:PrincipalTag/x509Subject/<field>` (eg CN maps to `PrincipalTag: aws:PrincipalTag/x509Subject/CN`)
+- Issuer fields as `PrincipalTag: aws:PrincipalTag/x509Issuer/<field>`
+- Subject Alternative Name (SAN) as `PrincipalTag: aws:PrincipalTag/x509SAN/<field>`
+
+Of the above, only some of them will be set by Teleport:
+- Subject Common Name: Teleport user that requested the access
+- Issuer Common Name: Teleport cluster name
+- Not After: this will be set based on the user's current session expiration
+
+### AWS Session creation
+After generating the certificate, Teleport will call `rolesanywhere.CreateSession` and exchange the certificate for AWS credentials.
```

Comment:
Can we be more precise than Teleport here? Will this be the Auth Server or the client?